hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainerMetadataInspector.java

```diff
@@ -0,0 +1,309 @@
+          OzoneConsts.BLOCK_COMMIT_SEQUENCE_ID, messageBuilder);
+
+      // Count number of block keys and total bytes used in the DB.
+      long usedBytesTotal = 0;
+      long blockCountTotal = 0;
+      long pendingDeleteBlockCountTotal = 0;
+      // Count normal blocks.
+      try (BlockIterator<BlockData> blockIter =
+               store.getBlockIterator(
+                   MetadataKeyFilters.getUnprefixedKeyFilter())) {
+
+        while (blockIter.hasNext()) {
+          blockCountTotal++;
+          usedBytesTotal += getBlockLength(blockIter.nextBlock());
+        }
+      }
+
+      String schemaVersion =
+          ((KeyValueContainerData) containerData).getSchemaVersion();
+
+      // Count pending delete blocks.
+      if (schemaVersion.equals(OzoneConsts.SCHEMA_V1)) {
+        try (BlockIterator<BlockData> blockIter =
+                 store.getBlockIterator(
+                     MetadataKeyFilters.getDeletingKeyFilter())) {
+
+          while (blockIter.hasNext()) {
+            blockCountTotal++;
+            pendingDeleteBlockCountTotal++;
+            usedBytesTotal += getBlockLength(blockIter.nextBlock());
+          }
+        }
+      } else if (schemaVersion.equals(OzoneConsts.SCHEMA_V2)) {
+        DatanodeStoreSchemaTwoImpl schemaTwoStore =
+            (DatanodeStoreSchemaTwoImpl) store;
+        pendingDeleteBlockCountTotal =
+            countPendingDeletesSchemaV2(schemaTwoStore);
+      } else {
+        messageBuilder.append("Cannot process deleted blocks for unknown " +
+            "container schema ").append(schemaVersion);
+      }
+
+      // Count number of files in chunks directory.
+      countFilesAndAppend(containerData.getChunksPath(), messageBuilder);
+
+      messageBuilder.append("Container state: ")
+          .append(containerData.getState()).append("\n");
+      messageBuilder.append("Schema Version: ")
+          .append(schemaVersion).append("\n");
+      messageBuilder.append("Total block keys in DB: ").append(blockCountTotal)
+          .append("\n");
+      messageBuilder.append("Total used bytes in DB: ").append(usedBytesTotal)
+          .append("\n");
+      messageBuilder.append("Total pending delete block keys in DB: ")
+          .append(pendingDeleteBlockCountTotal).append("\n");
+
+      boolean blockCountPassed =
+          checkMetadataMatchAndAppend(metadataTable, OzoneConsts.BLOCK_COUNT,
+              blockCount,
+              blockCountTotal, messageBuilder);
+      boolean bytesUsedPassed =
+          checkMetadataMatchAndAppend(metadataTable,
+              OzoneConsts.CONTAINER_BYTES_USED,
+              bytesUsed, usedBytesTotal, messageBuilder);
+      passed = blockCountPassed && bytesUsedPassed;
+    } catch(IOException ex) {
+      LOG.error("Inspecting container {} failed",
+          containerData.getContainerID(), ex);
+    }
+
+    if (passed) {
+      LOG.trace(messageBuilder.toString());
+    } else {
+      LOG.error(messageBuilder.toString());
+    }
```

Comment:
Can we write to a different file instead of Datanode log. It will be easier to debug.

Reply:
Yeah I think that's a good idea. What are your thoughts on:
1. Separate file for each container vs. one file for all containers?
2. Json output vs. unstructured output?

Reply:
I think consolidating all container inspector reports in 1 file would be much easier to visualize and debug.

Though Json would definitely be easier to parse, I don't have strong opinion about either as it will be a one time report.

Reply:
After some offline discussion, I'm now sending the report to a log with a different name. Report output can be sent to a separate file with a log4j config if desired.